oak_idl_gen_services/src/lib.rs

```diff
@@ -282,7 +282,7 @@ fn generate_service_method(rpc_call: &RPCCall) -> Vec<String> {
     // implementation of this method, therefore needs to be wrapped in `oak_idl::Message` in order
     // to transfer its ownership to the caller.
     vec![format!(
-        "    fn {}(&self, request: &{}) -> oak_idl::utils::Message<{}>;",
+        "    fn {}(&self, request: &{}) -> Result<oak_idl::utils::Message<{}>, oak_idl::Error>;",
```

Comment:
This may need to be `core::Result`, right?

Reply:
Doesn't seem like it from my testing, I think Vec was unique because it's a part of alloc.